### ISSUE TITLE
Training from a merged chcekpoint achieves zero training accuracy after the first step

### ISSUE THREAD

**Issue by l-k-11235** [eole-nlp/eole#64]
Here is the beginning of the log:
```shell
[2024-07-10 15:58:00,163 INFO] Loading checkpoint from ./finetune/merged_step_2000
[2024-07-10 15:58:00,166 WARNING] You have a CUDA device, should run with -gpu_ranks
[2024-07-10 15:58:00,234 INFO] Option: report_every, value: 1, overriding model: 50
[2024-07-10 15:58:00,234 INFO] Option: transforms, value: ['insert_mask_before_placeholder', 'onmt_tokenize', 'filtertoolong'], overriding model: ['filtertoolong']
[2024-07-10 15:58:00,234 INFO] Option: save_data, value: ./finetune/llama3-8b-instruct-finetune, overriding model: None
[2024-07-10 15:58:00,234 INFO] Option: src_vocab, value: /nas-labs/LM/models/llama3/llama3-8b-instruct/vocab.txt, overriding model: None
[2024-07-10 15:58:00,234 INFO] Option: tensorboard_log_dir, value: ./finetune/llama3-8b-instruct-finetune/logs/, overriding model: runs/eole
[2024-07-10 15:58:00,234 INFO] Option: seed, value: 1234, overriding model: -1
[2024-07-10 15:58:00,234 INFO] Option: tensorboard, value: True, overriding model: False
[2024-07-10 15:58:00,234 INFO] Option: default_specials, value: [], overriding model: ['<unk>', '<blank>', '<s>', '</s>']
[2024-07-10 15:58:00,234 INFO] Option: batch_size, value: 1, overriding model: 896
[2024-07-10 15:58:00,234 INFO] Option: batch_type, value: sents, overriding model: tokens
[2024-07-10 15:58:00,234 INFO] Option: train_steps, value: 10, overriding model: 100000
[2024-07-10 15:58:00,234 INFO] Option: lora_layers, value: ['linear_values', 'linear_query', 'linear_keys', 'final_linear'], overriding model: []
[2024-07-10 15:58:00,234 INFO] Option: param_init_glorot, value: True, overriding model: False
[2024-07-10 15:58:00,234 INFO] Option: valid_batch_size, value: 1, overriding model: 256
[2024-07-10 15:58:00,234 INFO] Option: bucket_size, value: 10, overriding model: 262144
[2024-07-10 15:58:00,234 INFO] Option: warmup_steps, value: 100, overriding model: 4000
[2024-07-10 15:58:00,234 INFO] Option: keep_checkpoint, value: 100, overriding model: -1
[2024-07-10 15:58:00,234 INFO] Option: lora_alpha, value: 4, overriding model: 1
[2024-07-10 15:58:00,234 INFO] Option: lora_dropout, value: 0.05, overriding model: 0.0
[2024-07-10 15:58:00,234 INFO] Option: save_checkpoint_steps, value: 100, overriding model: 5000
[2024-07-10 15:58:00,234 INFO] Option: zero_out_prompt_loss, value: True, overriding model: False
[2024-07-10 15:58:00,234 INFO] Option: num_workers, value: 1, overriding model: 2
[2024-07-10 15:58:00,234 INFO] Option: learning_rate, value: 2e-05, overriding model: 1.0
[2024-07-10 15:58:00,234 INFO] Option: gpu_ranks, value: [0], overriding model: []
[2024-07-10 15:58:00,234 INFO] Option: accum_count, value: [16], overriding model: [32]
[2024-07-10 15:58:00,234 INFO] Option: attention_dropout, value: [0.0], overriding model: [0.1]
[2024-07-10 15:58:00,234 INFO] Option: valid_steps, value: 3, overriding model: 10000
[2024-07-10 15:58:00,235 INFO] Option: dropout, value: [0.0], overriding model: [0.3]
[2024-07-10 15:58:00,235 INFO] Option: max_grad_norm, value: 0.0, overriding model: 5.0
[2024-07-10 15:58:00,235 INFO] Option: train_from, value: ./finetune/merged_step_2000, overriding model: None
[2024-07-10 15:58:00,235 INFO] Option: adam_beta2, value: 0.998, overriding model: 0.999
[2024-07-10 15:58:00,235 INFO] Option: model_path, value: ./finetune/llama3-8b-instruct-finetune, overriding model: model
[2024-07-10 15:58:00,235 INFO] Option: param_init, value: 0.0, overriding model: 0.1
[2024-07-10 15:58:00,235 INFO] Option: response_patterns, value: ['Voici un résumé en français du transcript audio :｟newline｠｟newline｠'], overriding model: ['Response : ｟newline｠']
[2024-07-10 15:58:00,235 INFO] Option: gpt2_pretok, value: True, overriding model: False
[2024-07-10 15:58:00,235 INFO] Option: src_subword_model, value: /nas-labs/LM/models/llama3/llama3-8b-instruct/bpe.model, overriding model: None
[2024-07-10 15:58:00,235 INFO] Option: tgt_subword_model, value: /nas-labs/LM/models/llama3/llama3-8b-instruct/bpe.model, overriding model: None
[2024-07-10 15:58:00,235 INFO] Option: tgt_subword_type, value: bpe, overriding model: none
[2024-07-10 15:58:00,235 INFO] Option: src_subword_type, value: bpe, overriding model: none
[2024-07-10 15:58:00,236 INFO] Default transforms (might be overridden downstream): ['insert_mask_before_placeholder', 'onmt_tokenize', 'filtertoolong'].
[2024-07-10 15:58:00,236 INFO] Parsed 2 corpora from -data.
[2024-07-10 15:58:00,236 INFO] Get special vocabs from Transforms: {'src': [], 'tgt': []}.
[2024-07-10 15:58:00,570 INFO] Transforms applied: ['onmt_tokenize', 'filtertoolong', 'insert_mask_before_placeholder']
[2024-07-10 15:58:00,610 INFO] The first 10 tokens of the vocabs are:['!', '"', '#', '$', '%', '&', "'", '(', ')', '*']
[2024-07-10 15:58:00,610 INFO] The decoder start token is: <s>
[2024-07-10 15:58:00,610 INFO] Default transforms (might be overridden downstream): ['insert_mask_before_placeholder', 'onmt_tokenize', 'filtertoolong'].
[2024-07-10 15:58:00,611 INFO] Parsed 2 corpora from -data.
[2024-07-10 15:58:00,611 INFO] Building model...
[2024-07-10 15:58:00,777 INFO] Adding LoRa layers for linear_values quant None
[2024-07-10 15:58:00,788 INFO] Adding LoRa layers for linear_query quant None
[2024-07-10 15:58:00,800 INFO] Adding LoRa layers for linear_keys quant None
[2024-07-10 15:58:00,814 INFO] Adding LoRa layers for final_linear quant None
[2024-07-10 15:58:00,830 INFO] Switching model to half() for FusedAdam legacy
[2024-07-10 15:58:00,830 INFO] Non quantized layer compute is fp16
[2024-07-10 15:58:03,556 INFO] DecoderModel(
  (decoder): TransformerLMDecoder(
    (transformer_layers): ModuleList(
      (0-31): 32 x TransformerLMDecoderLayer(
        (input_layernorm): RMSNorm()
        (self_attn): SelfMHA(
          (linear_keys): QLoraLinear_cls(
            in_features=4096, out_features=1024, bias=False
            (lora_dropout): Dropout(p=0.05, inplace=False)
          )
          (linear_values): QLoraLinear_cls(
            in_features=4096, out_features=1024, bias=False
            (lora_dropout): Dropout(p=0.05, inplace=False)
          )
          (linear_query): QLoraLinear_cls(
            in_features=4096, out_features=4096, bias=False
            (lora_dropout): Dropout(p=0.05, inplace=False)
          )
          (softmax): Softmax(dim=-1)
          (dropout): Dropout(p=0.0, inplace=False)
          (final_linear): QLoraLinear_cls(
            in_features=4096, out_features=4096, bias=False
            (lora_dropout): Dropout(p=0.05, inplace=False)
          )
        )
        (dropout): Dropout(p=0.0, inplace=False)
        (post_attention_layernorm): RMSNorm()
        (mlp): MLP(
          (gate_up_proj): Linear(in_features=4096, out_features=14336, bias=False)
          (down_proj): Linear(in_features=14336, out_features=4096, bias=False)
          (dropout_1): Dropout(p=0.0, inplace=False)
          (dropout_2): Dropout(p=0.0, inplace=False)
          (up_proj): Linear(in_features=4096, out_features=14336, bias=False)
        )
      )
    )
    (layer_norm): RMSNorm()
  )
  (tgt_emb): Embeddings(
    (embeddings): Embedding(128256, 4096, padding_idx=188)
    (dropout): Dropout(p=0.0, inplace=False)
  )
  (generator): Linear(in_features=4096, out_features=128256, bias=True)
)
[2024-07-10 15:58:03,562 INFO] embeddings: 525336576
[2024-07-10 15:58:03,562 INFO] encoder: 0
[2024-07-10 15:58:03,562 INFO] decoder: 6981292032
[2024-07-10 15:58:03,562 INFO] generator: 525464832
[2024-07-10 15:58:03,562 INFO] other: 0
[2024-07-10 15:58:03,562 INFO] * number of parameters: 8032093440
[2024-07-10 15:58:03,564 INFO] Trainable parameters = {'torch.float32': 0, 'torch.float16': 1703936, 'torch.uint8': 0, 'torch.int8': 0}
[2024-07-10 15:58:03,564 INFO] Non trainable parameters = {'torch.float32': 0, 'torch.float16': 8030389504, 'torch.uint8': 0, 'torch.int8': 0}
[2024-07-10 15:58:03,564 INFO]  * src vocab size = 128256
[2024-07-10 15:58:03,564 INFO]  * tgt vocab size = 128256
/usr/local/lib/python3.10/dist-packages/apex/contrib/optimizers/fp16_optimizer.py:58: UserWarning: The torch.cuda.*DtypeTensor constructors are no longer recommended. It's best to use methods such as torch.tensor(data, dtype=*, device='cuda') to create tensors. (Triggered internally at ../torch/csrc/tensor/python_tensor.cpp:78.)
  self.overflow_buf = torch.cuda.IntTensor([0])
[2024-07-10 15:58:04,151 INFO] Transforms applied: ['insert_mask_before_placeholder', 'onmt_tokenize', 'filtertoolong']
[2024-07-10 15:58:04,209 INFO] Default transforms (might be overridden downstream): ['insert_mask_before_placeholder', 'onmt_tokenize', 'filtertoolong'].
[2024-07-10 15:58:04,209 INFO] Parsed 2 corpora from -data.
[2024-07-10 15:58:04,225 INFO] Starting training on GPU: [0]
[2024-07-10 15:58:04,225 INFO] Start training loop and validate every 3 steps...
[2024-07-10 15:58:04,225 INFO] Scoring with: {'insert_mask_before_placeholder': InsertMaskBeforePlaceholdersTransform(), 'onmt_tokenize': ONMTTokenizerTransform(share_vocab=True, src_subword_kwargs={'bpe_model_path': '/nas-labs/LM/models/llama3/llama3-8b-instruct/bpe.model', 'bpe_dropout': 0.0}, src_onmttok_kwargs={'mode': 'none'}, tgt_subword_kwargs={'bpe_model_path': '/nas-labs/LM/models/llama3/llama3-8b-instruct/bpe.model', 'bpe_dropout': 0.0}, tgt_onmttok_kwargs={'mode': 'none'}), 'filtertoolong': FilterTooLongTransform(src_seq_length=512, tgt_seq_length=512)}

This fp16_optimizer is designed to only work with apex.contrib.optimizers.*
To update, use updated optimizers with AMP.
[2024-07-10 15:58:06,770 INFO] Weighted corpora loaded so far:
			* train_dataset: 1
[2024-07-10 15:58:11,882 INFO] Step  1/   10; acc: 77.7; ppl:  2.58; xent: 0.95; aux: 0.000; lr: 1.00e+00; sents:      16; bsz:  409/ 120/ 1; 855/252 tok/s;      8 sec;
[2024-07-10 15:58:14,897 INFO] Step  2/   10; acc: 0.0; ppl: 1127991.12; xent: 13.94; aux: 0.000; lr: 1.00e+00; sents:      16; bsz:  403/ 100/ 1; 2139/532 tok/s;     11 sec;
```

**Comment by vince62s** [comment]
your lr=1 is incorrect maybe you need to reset the optimizer

**Comment by l-k-11235** [comment]
Indeed : However I have `learning_rate:  2e-05` in the config

**Comment by vince62s** [comment]
you need to reset the optimizer

**Comment by l-k-11235** [comment]
With  `reset_optim: "all"` it works fine, thanks you !
